bluetooth: controller: Add vendor operations to TICKER_USER_ULL_HIGH_OPS

mtpr-ot opened the pull request:
Added to allow vendor specific increase of user operation capacity for ULL_HIGH, to support queuing additional ticker operations.

Signed-off-by: Morten Priess <mtpr@oticon.com>

wopu-ot reviewed (approved):
LGTM

cvinayak commented on subsys/bluetooth/controller/ll_sw/ull.c:
```diff
@@ -61,7 +61,11 @@
+#if defined(CONFIG_BT_TICKER_COMPATIBILITY_MODE)
 #define TICKER_USER_ULL_HIGH_OPS (3 + 1)
+#else
+#define TICKER_USER_ULL_HIGH_OPS (4 + 1)
```
In the upstream, analyses max use of ticker API in ULL_HIGH is 3 (ex. directed adv to slave transition, there is call to ticker_stop for adv, ticker_stop for adv_stop, ticker_start for slave role). Am I missing something? Is this the failure noticeable in the upstream samples?

mtpr-ot replied:
Yeah, we should update the samples with ticker_priority_set - the new kid on the block.

cvinayak replied:
Make the added additional count be defined in vendor files, and a common define be added.
```#define TICKER_USER_ULL_HIGH_OPS (3 + TICKER_USER_ULL_HIGH_VENDOR_OPS + 1)```

This way, RAM usage is low for typical samples that dont do additional ticker API call.

mtpr-ot replied:
Then we would need to guard the use of ticker priority by making it conditionally available. Otherwise the ticker will (may) assert when trying to set the priority. I don't like adding an unrelated condition, which will make the system fail if configured incorrectly. Note that even if ticker priority may only be used by certain vendors, ticker priority is available for all...

mtpr-ot replied:
> #define TICKER_USER_ULL_HIGH_OPS (3 + TICKER_USER_ULL_HIGH_VENDOR_OPS + 1)

OK, I will do this.

mtpr-ot committed:
bluetooth: controller: Add vendor operations to TICKER_USER_ULL_HIGH_OPS

Added to allow vendor specific increase of user operation capacity for
ULL_HIGH, to support queuing additional ticker operations.

Signed-off-by: Morten Priess <mtpr@oticon.com>

cvinayak reviewed (approved):
LGTM
